cli/command_snapshot_list.go

```diff
@@ -133,25 +128,52 @@ func (c *commandSnapshotList) run(ctx context.Context, rep repo.Repository) erro
 	}
 
 	if c.jo.jsonOutput {
-		for _, snapshotGroup := range snapshot.GroupBySource(manifests) {
-			snapshotGroup = snapshot.SortByTime(snapshotGroup, c.reverseSort)
+		return c.outputJSON(ctx, rep, manifests)
+	}
 
-			if c.maxResultsPerPath > 0 && len(snapshotGroup) > c.maxResultsPerPath {
-				snapshotGroup = snapshotGroup[len(snapshotGroup)-c.maxResultsPerPath:]
-			}
+	return c.outputManifestGroups(ctx, rep, manifests, strings.Split(relPath, "/"))
+}
+
+// SnapshotManifest defines the JSON output for the CLI snapshot commands.
+type SnapshotManifest struct {
+	*snapshot.Manifest
+	RetentionReasons []string `json:"retentionReason,omitempty"`
```

Comment:
I'm not sure I follow: why do we need an additional `RetentionReasons` here? It looks like the code below updates `Manifest.RetentionReasons`, and then copies it into `SnapshotManifest.RetentionReasons`. Could we just use the former directly? 
```diff
type Manifest struct {
  ...
-  RetentionReasons string[] `json:"-"`
+  RetentionReasons string[] `json:"retentionReason,omitempty"`

}
```
Is there anywhere else in the code that a `Manifest`'s JSON representation is used? Would such code break if we make the above change?

Reply:
The answer is here #1987: https://github.com/kopia/kopia/issues/1987#issuecomment-1140706429

> where the retention is NOT stored on purpose, since it needs to be dynamically computed.

Reply:
`snapshot.Manifest` is stored in the repo, that's how snapshots are tracked.
`snapshot.Manifest.RetentionReason` is explicitly excluded when the manifest is stored in the repo. The retention is dynamically computed based on policies, since the retention reason changes as time passes and new snapshots are added.

Reply:
Aha that makes sense–manifests are stored in JSON format